lib/async.js

```diff
@@ -144,7 +144,7 @@ module.exports = function resolve (x, opts, cb) {
                     pkg = opts.packageFilter(pkg, pkgfile);
                 }
                 
-                if (pkg.main) {
+                if (pkg.main && typeof pkg.main === 'string') {
```

Comment:
when would the "main" field be an object? Only the "browser" field has that as an option.

Reply:
Forgot the context, closing this.